tiledb/sm/array_schema/ndrectangle.h

```diff
@@ -138,6 +138,14 @@ class NDRectangle {
     return domain_;
   }
 
+  /**
```

Comment:
Let's make this more robust:
- Add checks before every usage of `domain_` in `ndrectangle.cc` that the pointer is not null. 
- Add comment on `domain_` declaration that it can be null

Reply:
Done, thanks!